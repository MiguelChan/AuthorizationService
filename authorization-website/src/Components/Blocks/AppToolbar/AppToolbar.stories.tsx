import React from 'react';
import {
  Story,
  Meta,
} from '@storybook/react';
import {
  AppToolbar,
} from './AppToolbar';
import { AppBar } from '@mui/material';

export default {
  title: 'Components/Blocks/AppToolbar',
  component: AppToolbar,
  decorators: [
    (Story) => (
      <AppBar position='static'>
        <Story />
      </AppBar>
    ),
  ],
} as Meta;

const Template: Story = () => <AppToolbar />;

export const Primary = Template.bind({});