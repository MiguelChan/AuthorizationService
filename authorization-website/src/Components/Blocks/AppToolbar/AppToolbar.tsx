import { Button, Toolbar, Typography } from '@mui/material';
import React from 'react';
import { Link } from 'react-router-dom';
import { ProfileContext, ProfileState } from '../../Context';

export const AppToolbar: React.FunctionComponent = () => {

  const profileContext: ProfileState = React.useContext(ProfileContext);

  const renderAddons = (): React.ReactElement => {
    const profile = profileContext.getProfile();
    
    if (profile === undefined) {
      return <Button color='inherit' component={Link} to='/login'>Login</Button>;
    }

    return (
      <Typography>
        Welcome back, {`${profile.firstName} ${profile.lastName}`}!
      </Typography>
    );
  };

  return (
    <Toolbar>
      <Typography variant='h6' component='div' sx={{ flexGrow: 1 }}>
        Auth-Hub
      </Typography>
      {renderAddons()}
    </Toolbar>
  );
};