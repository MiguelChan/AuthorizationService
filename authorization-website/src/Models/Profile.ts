export interface Profile {
  profileId: string;
  firstName: string;
  lastName: string;
  phoneNumber: string;
}

export interface ProfileDto {
  profile: Profile;
}