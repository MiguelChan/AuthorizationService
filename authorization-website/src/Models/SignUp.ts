export interface SignUpRequest {
  firstName: string;
  lastName: string;
  phoneNumber: string;
  emailAddress: string;
  password: string;
}

export interface SignUpResponse {
  profileId?: string;
  message?: string;
}