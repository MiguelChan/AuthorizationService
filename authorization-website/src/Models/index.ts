export * from './SignUpRequest';
export * from './Profile';