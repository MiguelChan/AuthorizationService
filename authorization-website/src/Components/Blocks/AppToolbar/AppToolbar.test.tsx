import React from 'react';
import {
  render,
  RenderResult,
} from '@testing-library/react';
import { AppToolbar } from './AppToolbar';
import { ProfileContext, ProfileState } from '../../Context';
import { AppBar } from '@mui/material';
import { Profile } from '../../../Models';
import { MemoryRouter } from 'react-router';

describe('AppToolbar', () => {

  it('Should render A LogIn Button when the User is not logged-in', () => {
    const profileState: ProfileState = {
      getProfile: () => undefined,
      setCurrentProfile: (): void => {},
    };

    const renderResult = setupComponent(profileState);
    expect(renderResult.container).toMatchSnapshot();
  });

  it('Should render the UserName in the ToolBar when user is logged-in', () => {
    const profile: Profile = {
      firstName: 'Miguel',
      lastName: 'Chan',
      phoneNumber: '',
      profileId: '',
    };

    const profileState: ProfileState = {
      getProfile: () => profile,
      setCurrentProfile: (): void => {},
    };

    const renderResult = setupComponent(profileState);
    expect(renderResult.container).toMatchSnapshot();
  });

  const setupComponent = (profileState: ProfileState): RenderResult => {
    const Component = (
      <MemoryRouter>
        <ProfileContext.Provider value={profileState}>
          <AppBar>
            <AppToolbar />
          </AppBar>
        </ProfileContext.Provider>
      </MemoryRouter>
    );

    return render(Component);
  };

});