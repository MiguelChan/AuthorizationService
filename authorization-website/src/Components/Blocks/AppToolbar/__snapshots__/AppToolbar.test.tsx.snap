// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppToolbar Should match the Snapshot 1`] = `
<div>
  <div
    class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
  >
    <div
      class="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
    >
      Auth-Hub
    </div>
    <button
      class="MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButtonBase-root  css-1y942vo-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Login
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
